core/src/main/java/com/graphhopper/routing/util/NameSimilarityEdgeFilter.java

```diff
@@ -46,34 +48,56 @@
  */
 public class NameSimilarityEdgeFilter implements EdgeFilter {
 
+    private static final Map<String, String> DEFAULT_REWRITE_MAP = new HashMap<String, String>() {{
+        // two char words will be ignored but ignore certain longer phrases (or rename them)
+        put("av.", "");
+        put("avenue", "");
+        put("avenida", "");
+        put("rd.", "");
+        put("road", "");
+        put("str.", "");
+        put("str", "");
+        put("straße", "");
+        put("strasse", "");
+        put("st.", ""); // saint vs. street
+        put("street", "");
```

Comment:
Rewriting to an empty string could be problematic in some more exotic cases. But I think the alternatives also introduce potential errors, so this is probably the best approach.

I miss a couple of strings here (not sure if they get abbreviated though, so maybe it's better not to add them?):
strada (e.g. Strada Provinciale 209)
via (e.g. Via Fosso Ancaiano)
calle (Calle el Dr. Mabuse)
plaza (Plaza de la Bámbola)

Some English ones:
lane
ln. (short form of lane)
place
pl.
track
tr.
pathway
path
crescend
cres
cres.
parade
pde

I found a great resource for street abbreviations in English [here](http://meteor.aihw.gov.au/content/index.phtml/itemId/270020). An even longer list can be found [here](http://meteor.aihw.gov.au/content/index.phtml/itemId/429387).

Reply:
Yes :)
We could even extract them from OSM:
https://github.com/komoot/photon/pull/46/files#diff-751e12c3a9da7a1987a479baab0ff97c
https://github.com/karussell/languator

But probably we should not overcomplicate things here and improve this just a bit.

Another question is how we handle those common words if they are part of the street name like with "Hauptstraße" we can't just remove them because e.g. `st` could be a part of a street name without the street-meaning

Reply:
> Another question is how we handle those common words if they are part of the street name like with "Hauptstraße" we can't just remove them because e.g. st could be a part of a street name without the street-meaning

Yes I think we cannot really do anything about these without introducing bigger errors as well.

> But probably we should not overcomplicate things here and improve this just a bit.

I know that English speakers love abbreviations, so I think fixing the English abbreviations is quite important. The names I mentioned above are pretty common, the ones in the first link are quite common as well, the second link contains more uncommon ones as well. But Google Maps loves to abbreviate them in english speaking countries as well, most abbreviations are rather short like Drive and Dr, so the dr will be removed anyway. So I think if anyone tries to combine the point_hint with GMaps Geocoding, this will become an issue.